src/main/kotlin/com/util/ReservationCalculator.kt

```diff
@@ -0,0 +1,37 @@
+package com.util
+
+import com.model.Reservation
+import com.parkingrate.ParkingRate
+import org.springframework.stereotype.Service
+import java.math.BigDecimal
+import java.time.Duration
+
+@Service
+class ReservationCalculator {
+    fun calculateReservationCost(reservation: Reservation,
```

Comment:
Ta funkcja działa raczej niepoprawnie. Kilka przykładów (dla typu Standard):
dla 1h zwrócona kwota to 1 - to jest ok
dla 2h zwrócona kwota to 2 - powinno być `1 + 2 = 3`
dla 3h zwrócona kwota to 3 - powinno być `1 + 2 + 1.5*2 = 6`
dla 4h zwrócna kwota to 4.5 - powinno być `1 + 2 + 1.5*2 + 1.5*1.5*2 = 10.5`

Powinieneś lepiej przetestować tę klasę.

Reply:
Racja, założenie aby zwracać jedynie stawkę za godzinę która minęła jest bez sensu. Powinienem trochę dłużej posiedzieć nad tą klasą i testami.